src/client/windshaft.js

```diff
@@ -122,19 +147,20 @@ export default class Windshaft {
 
     _intantiationChoices (metadata) {
         let choices = {
-            // default choices
-            backendFilters: true
+            backendFilters: true // default choices
         };
-        if (metadata) {
-            if (metadata.featureCount >= 0) {
-                choices.backendFilters = metadata.featureCount > MIN_FILTERING || !metadata.backendFiltersApplied;
-            }
+
+        if (metadata && metadata.featureCount >= 0) {
```

Comment:
Nice one! maybe just:

```js
const backendFilters = metadata && metadata.featureCount >= 0
    ? metadata.featureCount > MIN_FILTERING || !metadata.backendFiltersApplied
    : true

return { backendFilters }
```

Reply:
Thx! I'm gonna add a new explaining variable, but I think I'll keep the simple IF, to keep the current `shouldBeServerFiltered` variable